datalad/support/gitrepo.py

```diff
@@ -934,6 +934,24 @@ def clone(cls, url, path, *args, **kwargs):
                 gr.fsck()
             else:
                 lgr.warning("Experienced issues while cloning: %s", exc_str(fix_annex))
+        # make sure that Git doesn't mangle relative path specification into
+        # mildly obscure absolute paths
+        # https://github.com/datalad/datalad/issues/3538
+        if isinstance(url_ri, PathRI):
+            path = Path(url)
```

Comment:
I know this is inherited from 02e2b4c46 (BF: Avoid relpath mangling for submodule url configuration, 2020-01-10), but I'm not a fan of using `path` as the variable name.  The two main parameters for this method are `url` and `path`, so I find it confusing to redefine `path` with a value that is derived from `url`.  Perhaps `url_path`?

Reply:
Agree. `url_path` is fine.